### ISSUE TITLE
Improve error message on invalid SequenceParameter

### ISSUE THREAD

**Issue by nehaljwani** [conda/conda#5189]
For a malformed configuration parameter of type sequence:
```
(test) [hulk@ranganork conda]# cat ~/.condarc 
channels: defaults
```

The following shows the expected error:

```
(test) [hulk@ranganork conda]# conda config --validate 
InvalidTypeError: Parameter _channels = 'defaults' declared in <<merged>> has type <class 'str'>.
Valid types:
  - <class 'tuple'>
```

But the following throws an exception:

```
(test) [hulk@ranganork conda]# conda update conda
Traceback (most recent call last):
  File "/root/conda/conda/exceptions.py", line 632, in conda_exception_handler
    return_value = func(*args, **kwargs)
  File "/root/conda/conda/cli/main.py", line 134, in _main
    exit_code = args.func(args, p)
  File "/root/conda/conda/cli/main_update.py", line 65, in execute
    install(args, parser, 'update')
  File "/root/conda/conda/cli/install.py", line 128, in install
    context.validate_configuration()
  File "/root/conda/conda/common/configuration.py", line 844, in validate_configuration
    raise_errors(tuple(chain.from_iterable((errors, post_errors))))
  File "/root/conda/conda/common/configuration.py", line 842, in <genexpr>
    for name in self.parameter_names)
  File "/root/conda/conda/common/configuration.py", line 835, in _collect_validation_error
    func(*args, **kwargs)
  File "/root/conda/conda/common/configuration.py", line 448, in __get__
    result = typify_data_structure(self._merge(matches) if matches else self.default,
  File "/root/conda/conda/common/configuration.py", line 596, in _merge
    top_deduped = tuple(unique(concatv(top_lines, all_lines)))
  File "/root/conda/conda/_vendor/toolz/itertoolz.py", line 209, in unique
    for item in seq:
  File "/root/conda/conda/common/configuration.py", line 584, in <genexpr>
    top_lines = concat(get_marked_lines(m, ParameterFlag.top, self) for m in relevant_matches)
  File "/root/conda/conda/common/configuration.py", line 583, in get_marked_lines
    if flag is marker) if match else ()
TypeError: zip argument #2 must support iteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/conda/envs/test/bin/conda", line 11, in <module>
    load_entry_point('conda', 'console_scripts', 'conda')()
  File "/root/conda/conda/cli/main.py", line 162, in main
    return conda_exception_handler(_main, *args)
  File "/root/conda/conda/exceptions.py", line 636, in conda_exception_handler
    return handle_exception(e)
  File "/root/conda/conda/exceptions.py", line 626, in handle_exception
    print_unexpected_error_message(e)
  File "/root/conda/conda/exceptions.py", line 588, in print_unexpected_error_message
    stderrlogger.info(get_main_info_str(get_info_dict()))
  File "/root/conda/conda/cli/main_info.py", line 196, in get_info_dict
    channels = list(prioritize_channels(context.channels).keys())
  File "/root/conda/conda/base/context.py", line 448, in channels
    return self._channels
  File "/root/conda/conda/common/configuration.py", line 448, in __get__
    result = typify_data_structure(self._merge(matches) if matches else self.default,
  File "/root/conda/conda/common/configuration.py", line 596, in _merge
    top_deduped = tuple(unique(concatv(top_lines, all_lines)))
  File "/root/conda/conda/_vendor/toolz/itertoolz.py", line 209, in unique
    for item in seq:
  File "/root/conda/conda/common/configuration.py", line 584, in <genexpr>
    top_lines = concat(get_marked_lines(m, ParameterFlag.top, self) for m in relevant_matches)
  File "/root/conda/conda/common/configuration.py", line 583, in get_marked_lines
    if flag is marker) if match else ()
TypeError: zip argument #2 must support iteration
```

**Comment by nehaljwani** [comment]
Similar to #5184

**Comment by kalefranz** [comment]
Do you want to tackle this one @nehaljwani?

**Comment by kalefranz** [comment]
fixed with #5192
